include/deal.II/lac/full_matrix.templates.h

```diff
@@ -1663,7 +1663,9 @@ FullMatrix<number>::print_formatted (
   for (size_type i=0; i<m(); ++i)
     {
       for (size_type j=0; j<n(); ++j)
-        if (std::abs((*this)(i,j)) > threshold)
+        if (std::isnan(std::abs((*this)(i,j))))
```

Comment:
The problem is that complex numbers do not have an `isnan` method, so I work around it by taking the absolute value which will in any case be nan in case some of the ingredients are nan.